pinot-common/src/main/java/org/apache/pinot/common/utils/FileUploadDownloadClient.java

```diff
@@ -521,6 +523,28 @@ public SimpleHttpResponse uploadSegment(URI uri, String segmentName, InputStream
     return uploadSegment(uri, segmentName, inputStream, null, null, DEFAULT_SOCKET_TIMEOUT_MS);
   }
 
+  /**
```

Comment:
Not sure why we are having new APIs. Why not just add the headers in the existing APIs?

Reply:
changed to reuse the existing APIs. This involves more changes. Please check the recent commit.